torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -201,10 +200,48 @@ class ShapePropagator {
         continue;
       }
     }
-
     return tensor_types;
   }
 
+  c10::ScalarType typeToScalarType(c10::TypePtr type) {
+    if (auto cast = type->cast<CompleteTensorType>()) {
+      return cast->scalarType();
+    } else if(auto cast = type->cast<DimensionedTensorType>()) {
+      return cast->scalarType();
+    } else if (auto cast = type->cast<FloatType>()) {
+      return c10::ScalarType::Float;
+    } else if (auto cast = type->cast<IntType>()) {
+      return c10::ScalarType::Long;
```

Comment:
I'm not sure if this is correct, it's based on `CompleteTensorType::fromNumberType(TypePtr typ)` defined in `jit_type.h` which treats any subtype of IntType as a kLong. 
It's possible that the below boolean should be long as well, but I'm not sure why.